src/greynirseq/nicenlp/models/multiclass.py

```diff
@@ -169,16 +170,45 @@ def decode(self, tokens):
         # Remove the leading space, see 'encode' comment.
         return super().decode(tokens)[1:]
 
-    def predict_labels(self, sentence):
-        tokens = self.encode(sentence)
-        word_mask = torch.tensor([self.word_start_dict[t] for t in tokens.tolist()])
-        word_mask[0] = 0
-        word_mask[-1] = 0
-        word_mask = word_mask.unsqueeze(0)
-        tokens = tokens.unsqueeze(0)
+    def predict_labels(self, sentences: List[str], batch_size=1) -> Iterable[List[str]]:
```

Comment:
This returns a generator

Reply:
Generators er iterable (er það ekki?). Held hins vegar að týpan ætti að vera `Iterable[str]` nema ég sé að misskilja yield from.

Reply:
Já, generators eru iterable en þetta á að vera Iterable[str].

Reply:
Fair enough, but is there any reason not to use the generator type? https://docs.python.org/3/library/typing.html#typing.Generator

Reply:
Mér finnst það í þessu tilfelli vera eins og að skila "int" fyrir fall sem skilar "bool". Að skila Generator[List[str], None, None] finnst mér minna lýsandi.

Reply:
Það sem ykkur vantar er Iterator[], ekki Generator[].